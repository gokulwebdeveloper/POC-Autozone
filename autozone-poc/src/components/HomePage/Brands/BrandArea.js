import React from 'react';
import bgImgBrands1 from '../../../utlis/img/core-img/brand1.png';
import bgImgBrands2 from '../../../utlis/img/core-img/brand2.png';
import bgImgBrands3 from '../../../utlis/img/core-img/brand3.png';
import bgImgBrands4 from '../../../utlis/img/core-img/brand4.png';
import bgImgBrands5 from '../../../utlis/img/core-img/brand5.png';
import bgImgBrands6 from '../../../utlis/img/core-img/brand6.png';
import PropTypes from 'prop-types';

const BrandArea = props => {
    
    return (
        <div>
            <div className="brands-area d-flex align-items-center justify-content-between">
                <div className="single-brands-logo">
                    <img src={bgImgBrands1} alt="topshop" />
                </div>
                <div className="single-brands-logo">
                    <img src={bgImgBrands2} alt="mango" />
                </div>
                <div className="single-brands-logo">
                    <img src={bgImgBrands3} alt="zara" />
                </div>
                <div className="single-brands-logo">
                    <img src={bgImgBrands4} alt="bershka" />
                </div>
                <div className="single-brands-logo">
                    <img src={bgImgBrands5} alt="asos" />
                </div>
                <div className="single-brands-logo">
                    <img src={bgImgBrands6} alt="river island" />
                </div>
            </div>
        </div>
    );
};
export default BrandArea;