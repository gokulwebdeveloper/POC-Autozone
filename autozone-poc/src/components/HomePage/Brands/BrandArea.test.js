import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { configure, mount } from 'enzyme';
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import '../../../matchMedia.mock' // Must be imported before the tested file
import BrandArea from './BrandArea';

configure({ adapter: new Adapter() });

describe('BrandArea page testing', () => {
  let wrapper;
  //mount component for testing
  beforeEach(() => {
    wrapper = mount(<BrandArea />);
  })
  //Cleanup after test execution
  afterEach(() => {
    wrapper.unmount();
  });
  test('renders the img tag called', () => {
    const imgTag = wrapper.find("img");
    expect(imgTag).toHaveLength(6);
  });
  //SnapShot Match test
  it("SnapShot Match", () => {
    const brandAreaComponent = renderer.create(<BrandArea />).toJSON();
    expect(brandAreaComponent).toMatchSnapshot();
  });
})