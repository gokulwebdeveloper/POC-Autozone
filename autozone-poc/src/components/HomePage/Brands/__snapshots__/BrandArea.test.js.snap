// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandArea page testing SnapShot Match 1`] = `
<div>
  <div
    className="brands-area d-flex align-items-center justify-content-between"
  >
    <div
      className="single-brands-logo"
    >
      <img
        alt="topshop"
        src="brand1.png"
      />
    </div>
    <div
      className="single-brands-logo"
    >
      <img
        alt="mango"
        src="brand2.png"
      />
    </div>
    <div
      className="single-brands-logo"
    >
      <img
        alt="zara"
        src="brand3.png"
      />
    </div>
    <div
      className="single-brands-logo"
    >
      <img
        alt="bershka"
        src="brand4.png"
      />
    </div>
    <div
      className="single-brands-logo"
    >
      <img
        alt="asos"
        src="brand5.png"
      />
    </div>
    <div
      className="single-brands-logo"
    >
      <img
        alt="river island"
        src="brand6.png"
      />
    </div>
  </div>
</div>
`;
